src/riscv/lib/src/jit/builder/instr_map.rs

```diff
@@ -0,0 +1,144 @@
+use std::fmt::Debug;
+
+use crate::jit::builder::instruction::LoweredInstruction;
+use crate::jit::builder::sequence::sequence_analysis::InstrInfo;
+use crate::machine_state::memory::Address;
+
+/// A holder of the Addresses of the instructions in a sequence.
+#[derive(Debug, Clone)]
+pub(super) struct AddrMap {
+    addresses: Vec<Address>,
+}
+
+impl AddrMap {
+    /// Create a new, empty AddrMap.
+    pub(super) fn new() -> Self {
+        Self {
+            addresses: Vec::new(),
+        }
+    }
+
+    /// Insert a new address into the map.
+    pub(super) fn insert(&mut self, address: Address) {
+        self.addresses.push(address);
+    }
+
+    /// Retrieve the address at the provided instr ID.
+    pub(super) fn get(&self, instr_id: InstrId) -> Address {
+        self.addresses[instr_id.0]
+    }
+
+    /// Get the instr ID of the instruction at the given address.
+    pub(super) fn get_instr_id(&self, address: Address) -> Option<InstrId> {
+        self.addresses.binary_search(&address).ok().map(InstrId)
+    }
+}
+
+/// A holder of the instructions in a sequence with quick lookup via instr ID or address.
+#[derive(Debug, Clone)]
+pub(super) struct InstrMap<V> {
+    instructions: Vec<V>,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(super) struct InstrId(usize);
```

Comment:
Lets not intersperse type definitions between another type and its `impl` block.